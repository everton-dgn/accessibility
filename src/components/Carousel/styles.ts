import Layouts from 'layouts'
import styled, { css } from 'styled-components'

export const Wrapper = styled(Layouts.Col)`
  opacity: 0;
  animation: ${({ theme }) => theme.animations.fadeInMove('0', '30px', '0')}
    0.8s ease forwards;

  .slick-list,
  .slick-slider,
  .slick-track {
    position: relative;
    display: block;
  }

  .slick-loading .slick-slide,
  .slick-loading .slick-track {
    visibility: hidden;
  }

  .slick-slider {
    box-sizing: border-box;
    -webkit-user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    user-select: none;
    -webkit-touch-callout: none;
    -khtml-user-select: none;
    -ms-touch-action: pan-y;
    touch-action: pan-y;
    -webkit-tap-highlight-color: transparent;
    overflow: hidden;
    height: 46rem;
  }

  .slick-list {
    overflow: hidden;
    margin: 0;
    padding: 0;
  }

  .slick-list:focus {
    outline: 0;
  }

  .slick-list.dragging {
    cursor: pointer;
  }

  .slick-slider .slick-list,
  .slick-slider .slick-track {
    -webkit-transform: translate3d(0, 0, 0);
    -moz-transform: translate3d(0, 0, 0);
    -ms-transform: translate3d(0, 0, 0);
    -o-transform: translate3d(0, 0, 0);
    transform: translate3d(0, 0, 0);
  }

  .slick-track {
    top: 0;
    left: 0;
  }

  .slick-track:after,
  .slick-track:before {
    display: table;
    content: '';
  }

  .slick-track:after {
    clear: both;
  }

  .slick-slide {
    display: none;
    float: left;
    height: 100%;
    min-height: 1px;
    padding: 0 10px;
  }

  .slick-list {
    margin: 0 -1rem;
  }

  [dir='rtl'] .slick-slide {
    float: right;
  }

  .slick-slide img {
    display: block;
  }

  .slick-slide.slick-loading img {
    display: none;
  }

  .slick-slide.dragging img {
    pointer-events: none;
  }

  .slick-initialized .slick-slide {
    display: block;
  }

  .slick-vertical .slick-slide {
    display: block;
    height: auto;
    border: 1px solid transparent;
  }

  .slick-arrow.slick-hidden {
    display: none;
  }

  .slick-dots,
  .slick-next,
  .slick-prev {
    position: absolute;
    display: block;
    padding: 0;
  }

  .slick-next,
  .slick-prev {
    font-size: 0;
    line-height: 0;
    top: calc(50% - 3rem);
    -webkit-transform: translate(0, -50%);
    -ms-transform: translate(0, -50%);
    transform: translate(0, -50%);
    cursor: pointer;
    color: transparent;
    border: none;
    outline: 0;
    background-color: rgba(255, 255, 255, 0.73);
    width: 4.1rem;
    height: 8.2rem;
    transition: background-color 0.2s linear;
    opacity: 0.3;
  }

  .slick-slider:hover .slick-next,
  .slick-slider:hover .slick-prev {
    opacity: 1;
  }

  .slick-next:focus,
  .slick-next:hover,
  .slick-prev:focus,
  .slick-prev:hover {
    color: transparent;
    outline: 0;
    background-color: ${({ theme }) => theme.colors.white};
    display: flex;
  }

  .slick-next:focus:before,
  .slick-next:hover:before,
  .slick-prev:focus:before,
  .slick-prev:hover:before {
    opacity: 1;
  }

  .slick-next.slick-disabled:before,
  .slick-prev.slick-disabled:before,
  .slick-arrow.slick-prev.slick-disabled,
  .slick-arrow.slick-next.slick-disabled {
    display: none !important;
  }

  .slick-arrow {
    border-width: 2px;
    border-color: transparent;
    border-style: solid;

    &:focus {
      opacity: 1;
      border-color: #334bc8;
    }
  }

  .slick-next:before,
  .slick-prev:before {
    line-height: 1;
    opacity: 0.75;
  }

  .slick-prev {
    z-index: ${({ theme }) => theme.layers.base};
    left: 0;
    border-radius: 0 4rem 4rem 0;
  }

  [dir='rtl'] .slick-prev {
    right: -25px;
    left: auto;
  }

  .slick-prev:before {
    content: url("data:image/svg+xml,%3Csvg width='9' height='16' viewBox='0 0 9 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M7.75 14.5L0.999999 7.75L7.75 1' stroke='%23011FBB' stroke-width='2' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E");
  }

  .slick-next:before,
  [dir='rtl'] .slick-prev:before {
    content: url("data:image/svg+xml,%3Csvg width='10' height='16' viewBox='0 0 10 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1.625 1.25L8.375 8L1.625 14.75' stroke='%23011FBB' stroke-width='2' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
  }

  .slick-next {
    z-index: ${({ theme }) => theme.layers.base};
    right: 0;
    border-radius: 4rem 0 0 4rem;
  }

  [dir='rtl'] .slick-next {
    right: auto;
    left: -25px;
  }

  .slick-dotted.slick-slider {
    margin-bottom: 3rem;
  }

  .slick-dots {
    opacity: 0;
    animation: ${({ theme }) => theme.animations.fadeInMove('0', '30px', '0')}
      0.8s ease forwards;
    margin-top: ${({ theme }) => theme.spacings.xxxLarge};
    width: 100%;
    list-style: none;
    text-align: center;
  }

  .slick-dots li {
    position: relative;
    display: inline-block;
    width: 2rem;
    height: 2rem;
    margin: 0 0.5rem;
    padding: 0;
    cursor: pointer;
  }

  .slick-dots li button {
    font-size: 0;
    line-height: 0;
    display: block;
    width: 0.6rem;
    height: 0.6rem;
    padding: 0.5rem;
    cursor: pointer;
    color: transparent;
    border: 0;
    outline: 0;
    margin: auto;

    ${({ theme }) => css`
      background-color: ${theme.colors.bullet};
      border-radius: ${theme.border.circle};
    `}
  }

  .slick-dots li button {
    transition: 0.15s linear;
  }

  .slick-dots li button:focus,
  .slick-dots li button:hover {
    outline: 0;
    transform: scale(1.5);
  }

  .slick-dots li button:focus {
    box-shadow: 0 0 0 0.3rem #b0bbf6;
  }

  .slick-dots li button:focus:before,
  .slick-dots li button:hover:before {
    opacity: 1;
  }

  .slick-dots li.slick-active button {
    background-color: ${({ theme }) => theme.colors.primary};
  }
`
