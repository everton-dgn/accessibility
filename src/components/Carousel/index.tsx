import { ReactNode } from 'react'
import Slider from 'react-slick'
import * as S from 'components/Carousel/styles'

export type CarouselUsersCommentsProps = {
  children: ReactNode
}

const CarouselUsersComments = ({ children }: CarouselUsersCommentsProps) => {
  const settings = {
    dots: true,
    infinite: false,
    speed: 500,
    slidesToShow: 2,
    slidesToScroll: 2,
    accessibility: true,
    adaptiveHeight: true,
    responsive: [
      {
        breakpoint: 768,
        settings: {
          slidesToShow: 1,
          slidesToScroll: 1,
          dots: true
        }
      }
    ]
  }

  return (
    <S.Wrapper as="section" aria-labelledby="Carrossel">
      <Slider {...settings}>{children}</Slider>
    </S.Wrapper>
  )
}

export default CarouselUsersComments
