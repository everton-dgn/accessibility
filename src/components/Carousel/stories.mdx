import { Canvas, Meta, Story } from '@storybook/addon-docs'
import Carousel from '.'
import * as stories from './stories'

<Meta
  title="Design System/Carousel"
  component={Carousel}
  parameters={{
    controls: { sort: 'requiredFirst' }
  }}
/>

# Carousel

This is a carousel general purpose.

## Default

This component can receive data from an api and render an on-screen carousel.

The carousel has control by scroll arrows, navigation bulets and also by focus event in case of mouseless navigation.

On devices with screens greater than or equal to 768px, two cards are displayed per row in the carousel, on lower resolutions only one card per row is displayed.

To wait for data coming from an api a load is shown until everything can be loaded.

<Canvas>
  <Story story={stories.Default} />
</Canvas>
