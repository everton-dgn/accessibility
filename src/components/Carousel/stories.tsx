import { Story } from '@storybook/react'
import Layouts from 'layouts'
import styled, { css } from 'styled-components'
import Carousel from '.'
import * as C from 'components'
import { dataUsersComments } from 'components/CardUsersComments/mock'

const Container = styled(Layouts.Container).attrs({ as: 'section' })`
  margin: 2rem auto;
  padding: 0;

  ${({ theme }) => theme.breakPoint(800)} {
    margin: 12rem auto;
  }
`

const ContainerCarousel = styled(Layouts.Col)`
  max-width: 93vw;

  ${({ theme }) => css`
    margin-top: ${theme.spacings.large};

    ${theme.breakPoint(800)} {
      margin-top: ${theme.spacings.xtreme};
    }
  `}
`

const Template: Story = args => (
  <Container>
    <ContainerCarousel>
      <Carousel {...args}>
        {dataUsersComments.map(
          (
            {
              userName,
              city,
              state,
              urlUserBg,
              urlAvatar,
              urlProfile,
              comment
            },
            i
          ) => (
            <C.CardUsersComments
              key={userName + i}
              userName={userName}
              city={city}
              state={state}
              urlUserBg={urlUserBg}
              urlAvatar={urlAvatar}
              urlProfile={urlProfile}
              heigthREM={38}
              comment={comment}
            />
          )
        )}
      </Carousel>
    </ContainerCarousel>
  </Container>
)

export const Default = Template.bind({})
