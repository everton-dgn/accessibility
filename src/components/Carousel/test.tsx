import { screen } from '@testing-library/react'
import Carousel from '.'
import { renderWithTheme } from 'utils/tests/helpers'

describe('<CarouselUsersComments />', () => {
  it('should render the carousel with children', () => {
    const { container } = renderWithTheme(
      <Carousel>
        <h1>Children exemplo</h1>
      </Carousel>
    )

    const children = screen.getByRole('heading', { name: 'Children exemplo' })

    expect(children).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
