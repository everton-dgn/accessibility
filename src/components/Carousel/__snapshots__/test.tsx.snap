// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselUsersComments /> should render the carousel with children 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
}

.c1 {
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards;
  animation: hngCSK 0.8s ease forwards;
}

.c1 .slick-list,
.c1 .slick-slider,
.c1 .slick-track {
  position: relative;
  display: block;
}

.c1 .slick-loading .slick-slide,
.c1 .slick-loading .slick-track {
  visibility: hidden;
}

.c1 .slick-slider {
  box-sizing: border-box;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  -webkit-touch-callout: none;
  -khtml-user-select: none;
  -ms-touch-action: pan-y;
  touch-action: pan-y;
  -webkit-tap-highlight-color: transparent;
  overflow: hidden;
  height: 46rem;
}

.c1 .slick-list {
  overflow: hidden;
  margin: 0;
  padding: 0;
}

.c1 .slick-list:focus {
  outline: 0;
}

.c1 .slick-list.dragging {
  cursor: pointer;
}

.c1 .slick-slider .slick-list,
.c1 .slick-slider .slick-track {
  -webkit-transform: translate3d(0,0,0);
  -moz-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  -o-transform: translate3d(0,0,0);
  -webkit-transform: translate3d(0,0,0);
  -ms-transform: translate3d(0,0,0);
  transform: translate3d(0,0,0);
}

.c1 .slick-track {
  top: 0;
  left: 0;
}

.c1 .slick-track:after,
.c1 .slick-track:before {
  display: table;
  content: '';
}

.c1 .slick-track:after {
  clear: both;
}

.c1 .slick-slide {
  display: none;
  float: left;
  height: 100%;
  min-height: 1px;
  padding: 0 10px;
}

.c1 .slick-list {
  margin: 0 -1rem;
}

.c1 [dir='rtl'] .slick-slide {
  float: right;
}

.c1 .slick-slide img {
  display: block;
}

.c1 .slick-slide.slick-loading img {
  display: none;
}

.c1 .slick-slide.dragging img {
  pointer-events: none;
}

.c1 .slick-initialized .slick-slide {
  display: block;
}

.c1 .slick-vertical .slick-slide {
  display: block;
  height: auto;
  border: 1px solid transparent;
}

.c1 .slick-arrow.slick-hidden {
  display: none;
}

.c1 .slick-dots,
.c1 .slick-next,
.c1 .slick-prev {
  position: absolute;
  display: block;
  padding: 0;
}

.c1 .slick-next,
.c1 .slick-prev {
  font-size: 0;
  line-height: 0;
  top: calc(50% - 3rem);
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  -webkit-transform: translate(0,-50%);
  -ms-transform: translate(0,-50%);
  transform: translate(0,-50%);
  cursor: pointer;
  color: transparent;
  border: none;
  outline: 0;
  background-color: rgba(255,255,255,0.73);
  width: 4.1rem;
  height: 8.2rem;
  -webkit-transition: background-color 0.2s linear;
  transition: background-color 0.2s linear;
  opacity: 0.3;
}

.c1 .slick-slider:hover .slick-next,
.c1 .slick-slider:hover .slick-prev {
  opacity: 1;
}

.c1 .slick-next:focus,
.c1 .slick-next:hover,
.c1 .slick-prev:focus,
.c1 .slick-prev:hover {
  color: transparent;
  outline: 0;
  background-color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c1 .slick-next:focus:before,
.c1 .slick-next:hover:before,
.c1 .slick-prev:focus:before,
.c1 .slick-prev:hover:before {
  opacity: 1;
}

.c1 .slick-next.slick-disabled:before,
.c1 .slick-prev.slick-disabled:before,
.c1 .slick-arrow.slick-prev.slick-disabled,
.c1 .slick-arrow.slick-next.slick-disabled {
  display: none !important;
}

.c1 .slick-arrow {
  border-width: 2px;
  border-color: transparent;
  border-style: solid;
}

.c1 .slick-arrow:focus {
  opacity: 1;
  border-color: #334bc8;
}

.c1 .slick-next:before,
.c1 .slick-prev:before {
  line-height: 1;
  opacity: 0.75;
}

.c1 .slick-prev {
  z-index: 10;
  left: 0;
  border-radius: 0 4rem 4rem 0;
}

.c1 [dir='rtl'] .slick-prev {
  right: -25px;
  left: auto;
}

.c1 .slick-prev:before {
  content: url("data:image/svg+xml,%3Csvg width='9' height='16' viewBox='0 0 9 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M7.75 14.5L0.999999 7.75L7.75 1' stroke='%23011FBB' stroke-width='2' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E");
}

.c1 .slick-next:before,
.c1 [dir='rtl'] .slick-prev:before {
  content: url("data:image/svg+xml,%3Csvg width='10' height='16' viewBox='0 0 10 16' fill='none' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M1.625 1.25L8.375 8L1.625 14.75' stroke='%23011FBB' stroke-width='2' stroke-linecap='round' stroke-linejoin='round'/%3E%3C/svg%3E%0A");
}

.c1 .slick-next {
  z-index: 10;
  right: 0;
  border-radius: 4rem 0 0 4rem;
}

.c1 [dir='rtl'] .slick-next {
  right: auto;
  left: -25px;
}

.c1 .slick-dotted.slick-slider {
  margin-bottom: 3rem;
}

.c1 .slick-dots {
  opacity: 0;
  -webkit-animation: hngCSK 0.8s ease forwards;
  animation: hngCSK 0.8s ease forwards;
  margin-top: 3.2rem;
  width: 100%;
  list-style: none;
  text-align: center;
}

.c1 .slick-dots li {
  position: relative;
  display: inline-block;
  width: 2rem;
  height: 2rem;
  margin: 0 0.5rem;
  padding: 0;
  cursor: pointer;
}

.c1 .slick-dots li button {
  font-size: 0;
  line-height: 0;
  display: block;
  width: 0.6rem;
  height: 0.6rem;
  padding: 0.5rem;
  cursor: pointer;
  color: transparent;
  border: 0;
  outline: 0;
  margin: auto;
  background-color: #EBEDFA;
  border-radius: 50%;
}

.c1 .slick-dots li button {
  -webkit-transition: 0.15s linear;
  transition: 0.15s linear;
}

.c1 .slick-dots li button:focus,
.c1 .slick-dots li button:hover {
  outline: 0;
  -webkit-transform: scale(1.5);
  -ms-transform: scale(1.5);
  transform: scale(1.5);
}

.c1 .slick-dots li button:focus {
  box-shadow: 0 0 0 0.3rem #b0bbf6;
}

.c1 .slick-dots li button:focus:before,
.c1 .slick-dots li button:hover:before {
  opacity: 1;
}

.c1 .slick-dots li.slick-active button {
  background-color: #334BC8;
}

<div>
  <section
    aria-labelledby="Carrossel"
    class="c0 c1"
  >
    <div
      class="slick-slider slick-initialized"
    >
      
      <div
        class="slick-list"
        style="height: 0px;"
      >
        <div
          class="slick-track"
          style="opacity: 1; transform: translate3d(0px, 0px, 0px);"
        >
          <div
            aria-hidden="false"
            class="slick-slide slick-active slick-current"
            data-index="0"
            style="outline: none; width: 0px;"
            tabindex="-1"
          >
            <div>
              <h1
                style="width: 100%; display: inline-block;"
                tabindex="-1"
              >
                Children exemplo
              </h1>
            </div>
          </div>
        </div>
      </div>
      
      
    </div>
  </section>
</div>
`;
